RiotSwiftUI/Modules/Authentication/VerifyEmail/Coordinator/AuthenticationVerifyEmailCoordinator.swift

```diff
@@ -0,0 +1,168 @@
+    
+    private var indicatorPresenter: UserIndicatorTypePresenterProtocol
+    private var loadingIndicator: UserIndicator?
+    
+    /// The authentication service used for the registration.
+    var authenticationService: AuthenticationService { parameters.authenticationService }
+    /// The wizard used to handle the registration flow.
+    var registrationWizard: RegistrationWizard { parameters.registrationWizard }
+    
+    private var currentTask: Task<Void, Error>? {
+        willSet {
+            currentTask?.cancel()
+        }
+    }
+    
+    // MARK: Public
+
+    // Must be used only internally
+    var childCoordinators: [Coordinator] = []
+    @MainActor var completion: ((AuthenticationVerifyEmailViewModelResult) -> Void)?
+    
+    // MARK: - Setup
+    
+    @MainActor init(parameters: AuthenticationVerifyEmailCoordinatorParameters) {
+        self.parameters = parameters
+        
+        let viewModel = AuthenticationVerifyEmailViewModel()
+        let view = AuthenticationVerifyEmailScreen(viewModel: viewModel.context)
+        authenticationVerifyEmailViewModel = viewModel
+        authenticationVerifyEmailHostingController = VectorHostingController(rootView: view)
+        authenticationVerifyEmailHostingController.vc_removeBackTitle()
+        authenticationVerifyEmailHostingController.enableNavigationBarScrollEdgeAppearance = true
+        
+        indicatorPresenter = UserIndicatorTypePresenter(presentingViewController: authenticationVerifyEmailHostingController)
+    }
+    
+    // MARK: - Public
+    
+    func start() {
+        MXLog.debug("[AuthenticationVerifyEmailCoordinator] did start.")
+        Task { await setupViewModel() }
+    }
+    
+    func toPresentable() -> UIViewController {
+        return self.authenticationVerifyEmailHostingController
+    }
+    
+    // MARK: - Private
+    
+    /// Set up the view model. This method is extracted from `start()` so it can run on the `MainActor`.
+    @MainActor private func setupViewModel() {
+        authenticationVerifyEmailViewModel.completion = { [weak self] result in
+            guard let self = self else { return }
+            MXLog.debug("[AuthenticationVerifyEmailCoordinator] AuthenticationVerifyEmailViewModel did complete with result: \(result).")
+            
+            switch result {
+            case .send(let emailAddress):
+                self.sendEmail(emailAddress)
+            case .resend:
+                self.resentEmail()
+            case .cancel:
+                #warning("Reset the flow.")
+            }
+        }
+    }
+    
+    /// Show an activity indicator whilst loading.
+    /// - Parameters:
+    ///   - label: The label to show on the indicator.
+    ///   - isInteractionBlocking: Whether the indicator should block any user interaction.
+    @MainActor private func startLoading(label: String = VectorL10n.loading, isInteractionBlocking: Bool = true) {
```

Comment:
We don't seem to be using nondefault parameters in this class and the method is private, so maybe they are not needed?

Reply:
Updated. Worth saying that this is how the template generates the function, so most screens will probably have this method in this format. Do you think it makes sense to simplify it in the template and instead people can expand on it if needed?

Reply:
Hm then we might have to re-evaluate how we use the templates. In my understanding templates should both speed up the creation of repeated classes as well as help attain some amount of consistency, but at the end of the day it is the final code that matters, and how self-evident it is when we read it a year from now (when the template origin is long obscured).